lib/compat/wordpress-6.6/compat.php

```diff
@@ -0,0 +1,27 @@
+<?php
+/**
+ * WordPress 6.6 compatibility functions.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Change the Patterns submenu link and remove the Template Parts submenu for
+ * the Classic theme. This function should not be backported to core, and should be
+ * removed when the required WP core version for Gutenberg is >= 6.6.0.
+ *
+ * @global array $submenu
+ */
+function gutenberg_change_patterns_link_and_remove_template_parts_submenu_item() {
+	if ( ! wp_is_block_theme() ) {
+		global $submenu;
+		foreach ( $submenu['themes.php'] as $key => $item ) {
```

Comment:
The index of the Template Parts submenu varies depending on the core version. So we need to explore the submenu with `foreach()` (See [this changeset](https://core.trac.wordpress.org/changeset/57543)).

Reply:
I noticed that this line produces PHP warnings.

```
[02-May-2024 08:18:23 UTC] PHP Warning:  Trying to access array offset on value of type null in ~/gutenberg/wp-content/plugins/gutenberg/lib/compat/wordpress-6.6/compat.php on line 18
[02-May-2024 08:18:23 UTC] PHP Warning:  foreach() argument must be of type array|object, null given in ~/gutenberg/wp-content/plugins/gutenberg/lib/compat/wordpress-6.6/compat.php on line 18
```